pkgs/by-name/in/installShellFiles/package.nix

```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  callPackage,
+  makeSetupHook,
+}:
+
+# Documentation can be found at:
+# - doc/hooks/installShellFiles.section.md at the root of Nixpkgs repo
+# - setup-hook.sh (with usage examples)
+
+let
+  setupHook = makeSetupHook { name = "install-shell-files"; } (
+    builtins.path {
+      name = "install-shell-files-hook";
+      path = ./setup-hook.sh;
+      recursive = false;
+      sha256 = "sha256-cLDb6u/2p82wecm4P9spdSDYvBWQv96DnUKXXdzAeYE=";
+    }
```

Comment:
I'm a bit leery of this `builtins.path` call. Why add it here, when the original didn't include it?

Reply:
It has some purposes.
- Reproducibility, as explained in https://nix.dev/guides/best-practices.html#reproducible-source-paths
- the `name` attribute can be useful to track it in the Nix Store and in the logs
- the hash assures the integrity of this file

I think it is a nice add-on for such an important file.

Reply:
I will remove.
Too much maintenance work.